airflow-core/src/airflow/dag_processing/manager.py

```diff
@@ -871,6 +871,7 @@ def _create_process(self, dag_file: DagFileInfo) -> DagFileProcessorProcess:
             id=id,
             path=dag_file.absolute_path,
             bundle_path=cast("Path", dag_file.bundle_path),
+            bundle_version=dag_file.bundle_version,
```

Comment:
what's the purpose of this? I thought that with dag file processing, we are always tracking the latest version and we don't specify it here intentionally -- and i'm not even sure where this argument goes to in the interface

Reply:
This is needed so that we specifically store dag_id and the version in dagbag.dags.

If you have two dagruns running, with different number of task instances, you will notice that when the Scheduler does `dag_run.update_state`, some task will be marked removed because the retried dag from dagbag has fewer tasks